packages/api/portfolioDrafts/funding/createFundingStep.ts

```diff
@@ -1,74 +1,41 @@
-  isValidUuidV4,
-  isClinNumber,
-  isFundingAmount,
-} from "../../utils/validation";
-
-export interface ClinValidationError {
-  clinNumber: string;
-  invalidParameterName: string;
-  invalidParameterValue: string;
-  validationMessage: ValidationMessage;
-}
-
-/**
- * Accepts a Funding Step and performs input validation
- * on all Clins contained therein.
- * @returns a collection of clin validation errors
- */
-export function validateFundingStepClins(fs: FundingStep): Array<ClinValidationError> {
-  return fs.task_orders
-    .flatMap((taskOrder) => taskOrder.clins)
-    .map(validateClin)
-    .reduce((accumulator, validationErrors) => accumulator.concat(validationErrors), []);
-}
+import { DATABASE_ERROR, NO_SUCH_PORTFOLIO_DRAFT } from "../../utils/errors";
+import schema = require("../../models/schema.json");
+import { ApiGatewayEventParsed } from "../../utils/eventHandlingTool";
+import { requestShapeValidation, businessRulesValidationForFundingStep } from "../../utils/requestValidation";
+import middy from "@middy/core";
+import jsonBodyParser from "@middy/http-json-body-parser";
+import validator from "@middy/validator";
+import JSONErrorHandlerMiddleware from "middy-middleware-json-error-handler";
+import cors from "@middy/http-cors";
+import xssSanitizer from "../xssSanitizer";
+import errorHandlingMiddleware from "../../utils/errorHandlingMiddleware";
+import { wrapSchema } from "../../utils/schemaWrapper";
 
 /**
  * Submits the Funding Step of the Portfolio Draft Wizard
  *
  * @param event - The POST request from API Gateway
  */
-export async function handler(event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> {
-  const portfolioDraftId = event.pathParameters?.portfolioDraftId;
-  if (!isPathParameterPresent(portfolioDraftId)) {
-    return PATH_PARAMETER_REQUIRED_BUT_MISSING;
-  }
-  if (!portfolioDraftId || !isValidUuidV4(portfolioDraftId)) {
-    return NO_SUCH_PORTFOLIO_DRAFT;
-  }
-  if (!event.body) {
-    return REQUEST_BODY_EMPTY;
-  }
-  if (!isValidJson(event.body)) {
-    return REQUEST_BODY_INVALID;
-  }
-  const requestBody = JSON.parse(event.body);
-  if (!isFundingStep(requestBody)) {
-    return REQUEST_BODY_INVALID;
-  }
-  const fundingStep: FundingStep = requestBody;
-  const errors: Array<ClinValidationError> = validateFundingStepClins(fundingStep);
-  if (errors.length) {
-    return createValidationErrorResponse({ input_validation_errors: errors });
-  }
+export async function baseHandler(
+  event: ApiGatewayEventParsed<FundingStep>,
+  context?: Context
+): Promise<APIGatewayProxyResult> {
+  // Perform shape validation
+  const setupResult = requestShapeValidation<FundingStep>(event);
+  if (setupResult instanceof SetupError) {
+    return setupResult.errorResponse;
+  }
+  const portfolioDraftId = setupResult.path.portfolioDraftId;
+  const fundingStep = event.body;
+  // Perform business rules validation
+  businessRulesValidationForFundingStep(fundingStep);
+  // Perform database call
```

Comment:
Do the code level comments add any context?  Is the code self-documenting enough without them?

Reply:
That's fair, I just had those comments to help myself organize the file. I'll take them out.